website/docs/src/content/cloud/storage/search.mdx

```diff
@@ -0,0 +1,487 @@
+        },
+        { role: "user", content: userMessage },
+      ],
+    });
+
+    // Store this interaction as a new memory
+    await StoreMemory({
+      userId,
+      memory: `User asked: "${userMessage}". I replied: "${response}"`,
+      importance: "medium",
+    });
+
+    return response;
+  },
+);
+```
+
+### Knowledge base search
+
+Another powerful application is a knowledge base with faceted search capabilities:
+
+```tsx
+const SearchKnowledgeBase = gensx.Component(
+  "SearchKnowledgeBase",
+  async ({ query, filters = {} }) => {
+    const search = await useSearch("knowledge-base");
+
+    // Generate embedding for the query
+    const embedding = await OpenAIEmbedding.run({
+      model: "text-embedding-3-small",
+      input: query,
+    });
+
+    // Build filter conditions from user-provided filters
+    const filterConditions = {};
+
+    if (filters.category) {
+      filterConditions.category = { $eq: filters.category };
+    }
+
+    if (filters.dateRange) {
+      filterConditions.publishedDate = {
+        $gte: filters.dateRange.start,
+        $lte: filters.dateRange.end,
+      };
+    }
+
+    if (filters.tags && filters.tags.length > 0) {
+      filterConditions.tags = { $containsAny: filters.tags };
+    }
+
+    // Perform hybrid search (vector + keyword) with filters
+    const results = await search.query({
+      vector: embedding.data[0].embedding,
+      text: query, // Also use keyword search
+      hybridAlpha: 0.3, // Balance between vector and keyword search
+      includeAttributes: true,
+      topK: 10,
+      filters:
+        Object.keys(filterConditions).length > 0
+          ? { where: filterConditions }
+          : undefined,
+    });
+
+    // Return formatted results
+    return results.map((result) => ({
+      title: result.attributes?.title,
+      snippet: result.attributes?.snippet,
+      url: result.attributes?.url,
+      category: result.attributes?.category,
+      tags: result.attributes?.tags,
+      score: result.score,
+    }));
+  },
+);
+```
+
+## Advanced usage
+
+### Filtering by metadata
+
+Use filters to narrow down search results:
+
+```tsx
+const search = await useSearch("articles");
+
+// Search with filters
+const results = await search.query({
+  vector: queryEmbedding,
+  filters: {
+    where: {
+      // Filter by category
+      category: { $eq: "sports" },
+      // Filter by date range
+      publishDate: {
+        $gte: "2023-01-01",
+        $lt: "2024-01-01",
+      },
+      // Filter by multiple possible authors
+      author: { $in: ["Alice", "Bob", "Carol"] },
+    },
+  },
+});
+```
+
+### Hybrid search
+
+Combine vector similarity with keyword search:
+
+```tsx
+const search = await useSearch("documents");
+
+const results = await search.query({
+  vector: queryEmbedding, // Vector for semantic search
+  text: "quarterly report", // Text for keyword search
```

Comment:
This example isn't correct. `text` isn't a query param. It needs to be specified as an array of strings in `rankBy`: https://turbopuffer.com/docs/fts

hybridAlpha is also not a thing